packages/webapp/src/pages/election/round-video-upload.tsx

```diff
@@ -0,0 +1,348 @@
+    useUALAccount,
+} from "_app";
+import {
+    Container,
+    Text,
+    Expander,
+    Heading,
+    Loader,
+    VideoSubmissionFormAndPreview,
+    VideoSubmissionPhase,
+    Button,
+    Link,
+} from "_app/ui";
+import {
+    CurrentElection,
+    ElectionStatus,
+    ErrorLoadingElection,
+    setElectionRoundVideo,
+    getRoundTimes,
+} from "elections";
+import { EncryptionPasswordAlert } from "encryption";
+import { EdenMember } from "members";
+import { RoundHeader } from "elections/components/ongoing-election-components";
+
+// TODO:
+// 1) add buttons that launch video upload service in new tab
+// 2) test inductions to ensure I haven't screwed up their video upload
+// 3) Test video uploads during (and after) an election
+
+export const RoundVideoUploadPage = () => {
+    const [isElectionComplete, setIsElectionComplete] = useState(false);
+    const {
+        data: currentElection,
+        isLoading: isLoadingElection,
+        isError: isErrorElection,
+    } = useCurrentElection();
+
+    useEffect(() => {
+        if (currentElection?.electionState === ElectionStatus.Registration) {
+            setIsElectionComplete(true);
+        }
+    }, [currentElection]);
+
+    const isLoading = isLoadingElection;
+    const isError = isErrorElection;
+
+    if (isLoading || !currentElection) return <LoaderSection />;
+    if (isError) return <ErrorLoadingElection />;
+
+    const renderBanner = true; // TODO: get end time for video submission
+
+    return (
+        <FluidLayout
+            title="Election-video upload service"
+            banner={
+                renderBanner && (
+                    <EncryptionPasswordAlert
+                        promptSetupEncryptionKey={
+                            currentElection?.electionState !==
+                            ElectionStatus.Registration
+                        }
+                    />
+                )
+            }
+        >
+            {isElectionComplete ? (
+                <div>Election is complete.</div>
+            ) : (
+                <div className="divide-y">
+                    <Container>
+                        <Heading size={1}>
+                            Election-video upload service
+                        </Heading>
+                    </Container>
+                    <Container className="space-y-6 pb-5">
+                        <Text>
+                            Election video files are typically large and require
+                            many minutes to upload completely. Do not close this
+                            tab during the upload. A confirmation will be
+                            displayed when your upload is complete, along with a
+                            preview of your video.
+                        </Text>
+                        <Text>
+                            Be patient, and remember you have 48 hours from the
+                            beginning of the election to complete your election
```

Comment:
question: Is this from the _beginning_ of the election or the _end_?

Reply:
(I don't have reason to believe think this is incorrect...just more of a "let's confirm this" sort of comment)

Reply:
It's the beginning, because that's the time that the contract tracks.  Also, I would calculating and displaying the actual deadline here.